frontend/setupChelonia.js

```diff
@@ -12,6 +12,52 @@ import manifests from './model/contracts/manifests.json'
 import { SETTING_CHELONIA_STATE, SETTING_CURRENT_USER } from './model/database.js'
 import { CHATROOM_USER_STOP_TYPING, CHATROOM_USER_TYPING, CHELONIA_STATE_MODIFIED, KV_EVENT, LOGIN_COMPLETE, LOGOUT, OFFLINE, ONLINE, RECONNECTING, RECONNECTION_FAILED, SERIOUS_ERROR } from './utils/events.js'
 
+const handleDeletedContract = async (contractID: string) => {
+  const { cheloniaState, contractState } = sbp('chelonia/contract/fullState', contractID)
+  if (!cheloniaState) return
+
+  await sbp('chelonia/contract/remove', contractID)
+
+  const rootGetters = sbp('state/vuex/getters')
+
+  switch (cheloniaState.type) {
+    case 'gi.contracts/chatroom': {
+      // TODO: Leave chatroom on our identity contract or our group, if we've
```

Comment:
Not having a chatroom->parent relationship makes this difficult to implement (efficiently, at least). We also need to add a new action to the `identity` contract, to delete chatrooms (currently, it's append-only).